src/python/WMCore/WorkQueue/WorkQueueReqMgrInterface.py

```diff
@@ -48,48 +47,53 @@ def __init__(self, **kwargs):
 
     def __call__(self, queue):
         """Synchronize WorkQueue and RequestManager"""
+        # ensure log records go to the correct logger object
+        queue.logger = self.logger
         msg = ''
-        try:  # pull in new work
-            self.logger.info("queueing new work")
-            work = self.queueNewRequests(queue)
-            msg += "New Work: %d\n" % work
+        try:  # Fetch data from ReqMgr and propagate work cancellation to workqueue
+            self.logger.info("Canceling aborted and force-completed requests")
+            count = self.cancelWork(queue)
+            msg += "Work canceled: %s, " % count
         except Exception as ex:
-            errorMsg = "Error caught during RequestManager pull"
-            self.logger.exception("%s: %s", errorMsg, str(ex))
+            self.logger.exception("Generic error while canceling work. Details: %s", str(ex))
 
-        try:  # get additional open-running work
-            self.logger.info("adding new element to open requests")
+        try:  # Close requests that no longer need to be open for new data
+            self.logger.info("Closing open requests")
+            workClosed = queue.closeWork()
```

Comment:
what will happen if this indeed will throw an exception. Does the queue will have orphan requests? How takes care of that? In other words how queue is managed. From the code changes it is unclear since it is passed as an input.